### ISSUE TITLE
Decentralized adaptive optimizers - build a testbed

### ISSUE THREAD

**Issue by justheuristic** [learning-at-home/hivemind#152]
We need a simple training configuration where we can test the effects of different strategies for decentralized adaptive optimizers

Known strategies
* [x] naive - average parameters and gradients
* [x] https://arxiv.org/abs/1901.09109
* [x] https://dl.acm.org/doi/10.1145/3412815.3416891
* [x] NDA paper 3 - see telegram

Implement
* [x] a basic training setup for CIFAR10 or wikitext2 (ping me @friday, we'll figure it out)
* [x] naive strategy with averaging parameters and optimizers
* [x] alternative: paper2 from above

Compare convergence with equal bandwidth load

**Comment by justheuristic** [comment]
Fixed in #243
